resources/design-document.md

```diff
@@ -0,0 +1,217 @@
+just having a Set<String(ingredientNames)>
+
+## 3. Use Cases
+
+---
+
+U1. As a customer, I want to create a new recipe
+
+U2. As a customer, I want to retrieve a specific recipe
+
+U3. As a customer, I want to be able to update a recipe with new values
+
+U4. As a customer, I want to create a list of ingredients that I can delete individual items from.
+
+U5. As a customer, I want variable units of measurement
+
+U6. As a customer, I want to be able to change amount of ingredients based on how many servings I want
+
+U7. As a customer, I want to sort recipes based on a certain ingredient
+
+U8. As a customer, I want to schedule meals ahead of time
+
+U9. As a customer, I want to pull recipes within a certain range of calories
+
+U10. As a customer, I want to compare recipes
+
+U11. As a customer, I want to save past lists
+
+U12. As a customer, I want to have my favorite recipes displayed on the homepage
+
+## 4. Project Scope
+
+---
+
+### 4.1 In Scope
+
+- Create recipes and display their names on the homepage
+- Display chosen recipe on its own page
+    - Toggle between imperial and metric system
+    - Allow for scaling ingredients according to number of servings
+- Add recipe ingredients to a list
+- Display list and quantities needed
+- Delete items from list
+
+
+### 4.2 Out of Scope
+
+- Ability to compare recipes
+- Scheduling meals ahead of time
+- Favorite recipes
+- Saving past lists
+- Filtering recipes by individual ingredients
+- Filtering recipes through a range of calories
+
+## 5. Proposed Architecture Overview
+
+---
+
+This app will provide the minimum viable product including creating, retrieving, updating, and deleting recipes,
+along with creating an editable shopping list, displaying all recipes on the homepage, and displaying a chosen
+recipe with adjustable requirements according to number of desired servings, and with the ability to toggle
+between systems of measurement.
+
+Current expected endpoints(`CreateRecipe`, `GetRecipe`, `GetAllRecipes`, `UpdateRecipe`, `DeleteRecipe`,  `CreateList`, `GetList`, `UpdateList`)
+
+Recipes and shopping lists will be stored in their respective DynamoDB tables. The homepage will display the 
+names of all recipes along with the option to edit, delete, navigate to a details page, or add the recipe to 
+the shopping list. The homepage will also have the option to create and save a new recipe to the table, and 
+navigate to the shopping list page.
+
+## 6. API
+
+---
+
+### 6.1 Public Models
+
+#### RecipeModel
+
+```
+String userId
+String recipeName
+Integer servings
+Map<Integer, String> recipeSteps
+List<Ingredient> ingredients
+Integer calories
+Enum systemOfMeasurement
+    - METRIC
+    - IMPERIAL
+```
+
+#### IngredientModel
+
+```
+String ingredientName
+Double amount
+Enum unitOfMeasurement
+    - OUNCES
+    - FLUID_OUNCES
+    - CUPS
+    - TEASPOONS
+    - TABLESPOONS
+    - GRAMS
+    - MILLIGRAMS
+    - MILLILITERS
+    - COUNT
+```
+
+#### ListModel
```

Comment:
I'd probably call this something like `ShoppingListModel` to avoid ambiguity of what type of list you're talking about.